sips/sip-012/sip-012-cost-limits-network-upgrade.md

```diff
@@ -0,0 +1,674 @@
+
+In the last few months, the [clarity-benchmarking](https://github.com/blockstack/clarity-benchmarking) project has done rigorous benchmarking on contemporary hardware to come up with more accurate cost-limits, with a focus on the `runtime` limits. The updated cost-limits are described in detail in [this forum post](https://forum.stacks.org/t/more-accurate-cost-functions-for-clarity-native-functions/12386).
+
+Any modification of cost-limits is a consensus-breaking change. There seems to be broad community support for changing the cost-limits, the question is exactly how and when they go into effect. A previous proposal suggested using a voting contract to determine the block height at which a network-upgrade, described in detail in [this Github discussion](https://github.com/blockstack/stacks-blockchain/discussions/2845). Unfortunately, this path would take at least 4 months in the best-case scenario.
+
+This SIP posits that the ongoing network congestion warrants a more expedient route to change the cost-limits, one that does not rely on an on-chain voting contract.
+
+
+# Specification
+
+## Assumptions
+
+This SIP is a method of last resort, considering the circumstances an exception is justified. All future network upgrades should use the voting contract (if appropriate); all hard-forks must follow the process described in SIP-000.
+
+## Proposal
+
+The Stacks Foundation or the governance group should choose a Bitcoin block height for the network upgrade. The block number should be at least 3 calendar weeks out from when this SIP transitions into “Accepted” state, so as to provide sufficient heads up to node operators.
+
+Miners, developers, Stackers and community members can demonstrate their support for this network upgrade in one of the following two ways:
+
+* _Send a contract-call transaction to indicate support for the upgrade_: The contract would aggregate the STX balance on the tx-sender account. It could additionally call into the PoX contract to separately aggregate the total Stacked STX amount. See the Appendix for an initial draft of such a contract.
+* _Send a BTC transaction to indicate support_: Many large STX holders are on multi-sig BTC wallets unable to issue contract-calls. Such wallets could instead send a pure BTC transaction to indicate their support for the vote; it would be a no-op for the Stacks chain, but can be verified off-chain see [Bitcoin support indication](#bitcoin-support-indication) for details.
+
+The SIP will be considered Recommended if wallets indicating support for the upgrade (through either mechanism) add up to > 10% of circulating supply of STX (approx 120M).
+
+In terms of how these cost-limits would actually be applied, this SIP proposes the following:
+* Add new functionality to stacks-blockchain that uses the current cost-limits by default and uses new cost-limits if the burn block height exceeds a configurable parameter (could be a compile time configuration to avoid runtime issues)
```

Comment:
"Add new functionality to the Stacks blockchain".

Reply:
Done